src/OutputWriters/netcdf_output_writer.jl

```diff
@@ -483,19 +483,8 @@ function NetCDFOutputWriter(model, outputs;
     # Ensure we can add any kind of metadata to the global attributes later by converting to Dict{Any, Any}.
     global_attributes = Dict{Any, Any}(global_attributes)
 
-    dataset, outputs, schedule = initialize_nc_file!(filepath,
-                                                     outputs,
-                                                     schedule,
-                                                     array_type,
-                                                     indices,
-                                                     with_halos,
-                                                     global_attributes,
-                                                     output_attributes,
-                                                     dimensions,
-                                                     overwrite_existing,
-                                                     deflatelevel,
-                                                     grid,
-                                                     model)
+    # Create empty temporal dataset in the first call.
+    dataset = Dataset(Base.Filesystem.tempname(),"c")
```

Comment:
@glwagner This is the only way I came up to be able to use the current `NetCDFOutputWriter` struct. I don't see an easy work around, but I also think it's not great to create an empty netCDF file each time we create a new instance of the `NetCDFOutputWriter`